test/DotNetty.Codecs.Tests/Base64Test.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Tests
+{
+    using System;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using DotNetty.Buffers;
+    using DotNetty.Codecs.Base64;
+    using Xunit;
+
+    public class Base64Test
+    {
+        [Fact]
+        public void TestNotAddNewLineWhenEndOnLimit()
+        {
+            var src = Unpooled.CopiedBuffer(Encoding.ASCII.GetBytes("____abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcde"));
+            var expectedEncoded = Unpooled.CopiedBuffer(Encoding.ASCII.GetBytes("YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXphYmNkZWZnaGlqa2xtbm9wcXJzdHV2d3h5emFiY2Rl"));
+            var buff = src.Slice(3, src.ReadableBytes - 3);
+            buff.ReadByte();
+            TestEncode(buff, expectedEncoded);
+        }
+
+        [Fact]
+        public void TestSimpleDecode()
```

Comment:
>TestSimpleDecode [](start = 20, length = 16)

would be great to cover a variety of test cases, e.g. lengths ranging from 0 to some number enough to wrap the line with NewLine. See [https://github.com/Azure/DotNetty/blob/dev/test/DotNetty.Handlers.Tests/TlsHandlerTest.cs#L63](https://github.com/Azure/DotNetty/blob/dev/test/DotNetty.Handlers.Tests/TlsHandlerTest.cs#L63) for example on how to do this.